### ISSUE TITLE
Add CMake logic to use FindPython instead of FindPythonInterp and FindPythonLibs

### ISSUE THREAD

**Issue by eirrgang** [kassonlab/gmxapi#150]
FindPythonInterp and FindPythonLibs are deprecated as of CMake 3.12 in favor of a new FindPython module in the style of modern CMake. It will be a while before we can count on it, but we can add logic to try to start using the new Find module.

(this issue is about as low priority as it could be...)
